### PR TITLE
perf(core): add option to remove blank text nodes from compiled templates

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This PR retrofits WS removal code against the 4.x.

This PR cherry-picks the following commits in order to retrofit WS removal from master to 4.x:
* d2c0d986d44f3a7a29bdeacbe973a5550f5c6839 
* 7d72d0eb9bbd4c197305dba0f0588fad3d592f59
* 8ea6c56fe1da08728316ed6c90ef8b0e89861447
* e03559867b2ccbc58b4ae0660f6cd84896ee199f
* https://github.com/angular/angular/pull/18995/commits/7d48d94abf23c6be15c92a12aa03e2eb1b887365

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
** ci/circleci: lint ** job is failing with:

> INVALID COMMIT MSG: "Revert "revert: revert: ci(aio): exclude changes in aio/content folder""
 => ERROR: The commit message does not match the format of "<type>(<scope>): <subject> OR revert: type(<scope>): <subject>"

This is not linked to this PR (guess it is failing on all the sha in this branch).

File: packages/compiler-cli/src/transformers/api.ts [hunk 1]
```diff
@@ -91,6 +91,10 @@ export interface CompilerOptions extends ts.CompilerOptions {
 
   // Whether to enable support for <template> and the template attribute (true by default)
   enableLegacyTemplate?: boolean;
+
+  // Whether to remove blank text nodes from compiled templates. It is `true` by default
+  // in Angular 5 and will be re-visited in Angular 6.
```
Comment (vicb) [inline packages/compiler-cli/src/transformers/api.ts]: Update here ?

Comment (pkozlowski-opensource) [inline packages/compiler-cli/src/transformers/api.ts]: Done

**Comment by mhevery** [comment]
Any progress on this?

**Comment by pkozlowski-opensource** [comment]
@mhevery for me this one is ready to be reviewed / merged (as it was reviewed when landing in master). Circle CI build failing is unrelated to this issue (see https://github.com/angular/angular/pull/18823#issuecomment-323980196).

I've aligned it with the code in master and rebased on top of 4.3x. Let me know if there is anything else I could do to help getting it in.

**Review by vicb — APPROVED** [review]

**Review by mhevery — APPROVED** [review]

**Comment by mhevery** [comment]
merged as b8b551cf2

**Pull request closed without merging**